pipeline/checksums/checksums.go

```diff
@@ -0,0 +1,64 @@
+package checksums
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/goreleaser/goreleaser/checksum"
+	"github.com/goreleaser/goreleaser/context"
+	"golang.org/x/sync/errgroup"
+)
+
+// Pipe for checksums
+type Pipe struct{}
+
+// Description of the pipe
+func (Pipe) Description() string {
+	return "Calculating checksums"
+}
+
+// Run the pipe
+func (Pipe) Run(ctx *context.Context) (err error) {
+	var g errgroup.Group
+	for _, artifact := range ctx.Artifacts {
```

Comment:
Yep. AFAIK when you `range list` it is the state of the list at that given time, later operation on it will not affect the loop.

e.g: https://play.golang.org/p/Xcr6YRkmGo